api/v1alpha1/executionhook_types.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ExecutionHookFinalizer = "executionhook.apps.x-k8s.io"
```

Comment:
Why is this needed?

Reply:
The finalizers ensure that our controller code is called to perform pre-delete hooks for the execution hook object.
The way delete of resources is handled is:
1. Perform pre-delete actions
2. Remove this finalizer
3. Return no error

More about this here https://book-v1.book.kubebuilder.io/beyond_basics/using_finalizers.html

Reply:
Is this finalizer generated automatically or added by you?  Please add comments to explain what it is used for as finalizer can have different purposes.

Reply:
I added this. Added comments explaining this.